src/journal/views.py

```diff
@@ -899,9 +900,19 @@ def article_file_make_galley(request, article_id, file_id):
     :return: a redirect to the URL at the GET parameter 'return'
     """
     article_object = get_object_or_404(submission_models.Article, pk=article_id)
-    file_object = get_object_or_404(core_models.File, pk=file_id)
+    janeway_file = get_object_or_404(core_models.File, pk=file_id)
+    blob = janeway_file.get_file(article_object, as_bytes=True)
+    content_file = ContentFile(blob)
+    content_file.name = janeway_file.original_filename
+
 
-    logic.create_galley_from_file(file_object, article_object, owner=request.user)
+    # Avoid circular import.
+    from production import logic as production_logic
+
+    production_logic.save_galley(
+        article_object, request, content_file,
+        is_galley=True,
+    )
```

Comment:
I tested this through the interface by uploading a manuscript file, converting it to a galley, and then checking the attributes in admin. It seems that there is no galley type being assigned. That's a problem, right? Is it possible there is still a logic error here?

Reply:
I tested with a PDF and a JATS file.

Reply:
was it a file with a valid mime that can be linked to a correct galley type? (PDF/HTML/XML)

Reply:
Just a note to record that this was an issue with my testing environment, and we confirmed the type is set properly.